src/components/timeRangePicker/TimeRangePicker.tsx

```diff
@@ -0,0 +1,302 @@
+import {
+  Interval,
+  isSameDay,
+  isMonday,
+  previousMonday,
+  nextMonday,
+} from "date-fns";
+import { Calendar } from "react-date-range";
+import { useClickOutside } from "../../hooks";
+import { Button } from "../button";
+import { Datepicker } from "../datepicker";
+import { Label } from "../input";
+import { SelectButton } from "../selectbutton";
+import {
+  Row,
+  UnifiedMultipleSelect,
+  IconButton,
+  DatepickerContainer,
+  CustomTimeRangeSelector,
+} from "./styles";
+import { DateOption, TimeRangeOption, TimeRangePickerProps } from "./types";
+
+const WEEKDAYS = [
+  "Monday",
+  "Tuesday",
+  "Wednesday",
+  "Thursday",
+  "Friday",
+  "Saturday",
+  "Sunday",
+];
+
+const DEFAULT_TIME_RANGE_OPTIONS: TimeRangeOption[] = [
+  {
+    id: "0",
+    start: "00:00",
+    end: "06:00",
+  },
+  {
+    id: "1",
+    start: "06:00",
+    end: "12:00",
+  },
+  {
+    id: "2",
+    start: "12:00",
+    end: "18:00",
+  },
+  {
+    id: "3",
+    start: "18:00",
+    end: "23:59",
+  },
+];
+
+const DEFAULT_TIME_RANGE: Interval = {
+  start: new Date(),
+  end: new Date(),
+};
+
+export function TimeRangePicker({
+  initTimeRange = DEFAULT_TIME_RANGE,
+  onChange,
+}: TimeRangePickerProps) {
+  const [selectedDate, setSelectedDate] = useState<Date>(
+    getInitSelectedDate(initTimeRange)
+  );
+  const [selectedTimeRange, setSelectedTimeRange] = useState(
+    DEFAULT_TIME_RANGE_OPTIONS[0]
+  );
+  const [dateOptions, setDateOptions] = useState<DateOption[]>(
+    getInitDateOptions(selectedDate)
+  );
+  const [isCalendarActive, setIsCalendarActive] = useState(false);
+
+  const showCustomTimeRange = selectedTimeRange.id === "custom";
+
+  const calendarWrapper = useRef<HTMLDivElement>(null);
+
+  useClickOutside(calendarWrapper, isCalendarActive, () => {
+    setIsCalendarActive(false);
+  });
+
+  // Update dateOptions when Monday of the selectedDate is changed
+  useEffect(() => {
+    setDateOptions(() => getDateOptions(getMonday(selectedDate)));
+  }, [getMonday(selectedDate).toLocaleDateString()]); // eslint-disable-line
+
+  // Submit timeRange change when selectedDate changed,
+  useEffect(() => {
+    submit();
+  }, [selectedDate.toLocaleDateString()]); // eslint-disable-line
+
+  // Submit timeRange change when selectedTimeRange changed,
+  // except for custom timeRange change, which will be handled separately
+  useEffect(() => {
+    if (selectedTimeRange.id !== "custom") {
+      submit();
+    }
+  }, [selectedTimeRange]); // eslint-disable-line
+
+  // Function to update dateOptions when arrows are clicked
+  const updateDateOptions = (direction: string) => {
+    if (direction === "backward") {
+      setDateOptions((prev) => {
+        return getDateOptions(previousMonday(new Date(prev[0].id)));
+      });
+    } else if (direction === "forward") {
+      setDateOptions((prev) => {
+        return getDateOptions(nextMonday(new Date(prev[0].id)));
+      });
+    }
```

Comment:
We can avoid addiotinal nesting

```suggestion
    }
    
    if (direction === "forward") {
      setDateOptions((prev) => {
        return getDateOptions(nextMonday(new Date(prev[0].id)));
      });
    }
```